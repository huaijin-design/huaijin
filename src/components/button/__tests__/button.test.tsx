import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Button } from '..';

describe('Button', () => {
  it('renders correctly', () => {
    const json = renderer.create(
      <Button>Default</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> primary', () => {
    const json = renderer.create(
      <Button primary>Default</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> ghost', () => {
    const json = renderer.create(
      <Button ghost>Ghost</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> disabled', () => {
    const json = renderer.create(
      <Button disabled>Disabled</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> text', () => {
    const json = renderer.create(
      <Button text>Text</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> circle', () => {
    const json = renderer.create(
      <Button circle>C</Button>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });
});
