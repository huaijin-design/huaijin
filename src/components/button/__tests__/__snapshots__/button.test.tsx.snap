// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

<button
  className="emotion-0"
>
  Default
</button>
`;

exports[`Button renders style rules -> circle 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  min-width: 32px;
  width: 32px;
  height: 32px;
  padding: 4px;
  overflow: hidden;
  border-radius: 50%;
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

<button
  className="emotion-0"
>
  C
</button>
`;

exports[`Button renders style rules -> disabled 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  box-shadow: none;
  cursor: not-allowed;
  background-color: rgba(112,112,112,0.12);
  color: rgba(112,112,112,0.26);
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

.emotion-0:hover {
  box-shadow: none;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0:active {
  box-shadow: none;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

<button
  className="emotion-0"
  disabled={true}
>
  Disabled
</button>
`;

exports[`Button renders style rules -> ghost 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  box-shadow: none;
  background: transparent;
  border: 1px solid #fff;
  color: #fff;
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

.emotion-0:hover {
  color: #707070;
  background-color: #fff;
}

.emotion-0:active {
  color: #707070;
  background-color: #fff;
}

<button
  className="emotion-0"
>
  Ghost
</button>
`;

exports[`Button renders style rules -> primary 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  background-color: #ff6f61;
  color: #fff;
  box-shadow: 0px 2px 4px rgba(255,111,97,0.75);
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(255,111,97,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(255,111,97,0.75);
}

<button
  className="emotion-0"
>
  Default
</button>
`;

exports[`Button renders style rules -> round 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  border-radius: 1.1428571428571428em;
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

<button
  className="emotion-0"
>
  Round
</button>
`;

exports[`Button renders style rules -> text 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  border: none;
  outline: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  min-width: 6.285714285714286em;
  min-height: 2.2857142857142856em;
  border-radius: 2px;
  padding: 5px 10px;
  -webkit-transition: box-shadow 0.3s ease,-webkit-transform 0.3s ease,background 0.3s ease;
  -webkit-transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  transition: box-shadow 0.3s ease,transform 0.3s ease,background 0.3s ease;
  font-size: 14px;
  background-color: #f5f5f5;
  color: #707070;
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
  box-shadow: none;
  background: none;
}

.emotion-0:hover {
  -webkit-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  -ms-transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
}

.emotion-0:active {
  -webkit-transform: translate3d(0,0,0) scale3d(1,1,1);
  -ms-transform: translate3d(0,0,0) scale3d(1,1,1);
  transform: translate3d(0,0,0) scale3d(1,1,1);
}

.emotion-0:hover {
  box-shadow: 0px 6px 12px rgba(222,222,222,0.75);
}

.emotion-0:active {
  box-shadow: 0px 2px 4px rgba(222,222,222,0.75);
}

<button
  className="emotion-0"
>
  Text
</button>
`;
